File: Leetcode/Go/2574.go
package main

import "fmt"

func leftRigthDifference(nums []int) []int {
    res := make([]int, len(nums))
    var rsum, lsum int

    for _, v := range nums {
        rsum += v
    }

    for i, v := range nums{
        rsum -= v
        res[i] = abs(lsum - rsum)
        lsum += v
    }

    return res
}

func abs(n int) int {
    if n < 0 {
        return -n
    }
    return n
}

func main() {
	nums := []int{2, 7, 11, 15}
	fmt.Println(nums)
	fmt.Print(leftRigthDifference(nums))
}

File: Leetcode/Python/2574.py
# Given a 0-indexed integer array nums, find a 0-indexed integer array answer where:

# answer.length == nums.length.
# answer[i] = |leftSum[i] - rightSum[i]|.
# Where:

# leftSum[i] is the sum of elements to the left of the index i in the array nums. If there is no such element, leftSum[i] = 0.
# rightSum[i] is the sum of elements to the right of the index i in the array nums. If there is no such element, rightSum[i] = 0.
# Return the array answer.
class Solution:
    def leftRigthDifference(self, nums: list[int]) -> list[int]:
        answer = []
        for i in range(len(nums)):
            answer.append(abs(sum(nums[:i])-sum(nums[i+1:])))
        return answer


print(Solution().leftRigthDifference([1]))
